webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/inventory/InventoryFileWatcher.scala

```diff
@@ -350,12 +349,40 @@ class ProcessFile(
    * a "file written" event.
    */
   type WatchQueue = ZQueue[Any, Nothing, Any, Nothing, WatchEvent, WatchEvent]
-  val queue = ZioRuntime.unsafeRun(Queue.bounded[WatchEvent](10000))
+  val watchEventQueue = ZioRuntime.unsafeRun(Queue.bounded[WatchEvent](16384))
+
+  /*
+   * We have a second queue that is used as buffer for the step between "file ok to be processed"
+   * and "saving file" because "saving file imply parsing XML and doing other memory and CPU
+   * inefficient.
+   * The buffer also allowed to filter inventory for identical nodes to avoid parsing old/not up to date
+   * inventory.
+   * As only a small structure is saved (SaveInventory is a name and two function to create stream from file), buffer can be big.
+   * When queue is full, we prefer to drop old inventories to new ones. In the worst case, they will be catched up by the
+   * scheduler.
+   */
+  protected val saveInventoryBuffer = ZQueue.sliding[(File, Option[File])](1024).runNow
+
+  /*
+   * The processing logic is to continually call sendToProcessorBlocking. When we
+   * take a file, we  de-duplicate files in buffer queue with the same name (ie: remove them).
+   * It likely means that we add several "whatch even: add" for that file (either copied several time, touched,
+   * or inotify event emission not what we thought)
+   */
+  val saveInventoryBufferProcessing = {
+    for {
+      inv <- saveInventoryBuffer.take
+      // deduplicate. TakeAll is not blocking
+      all <- saveInventoryBuffer.takeAll
+      _   <- saveInventoryBuffer.offerAll(all.filterNot { case (f, _) => inv._1.name == f.name })
```

Comment:
isn't there a risk that we loose inventory ?
Doc says:
   * For Sliding Queue: uses `Sliding` Strategy
   * If there is room in the queue, it places the values otherwise it removes the old elements and
   * enqueues the new ones. Always returns true.

Reply:
and it contradict the comment up there: if we reoffer old inventories, they become new